callbackPractice/callbackPractice.js

```diff
@@ -19,52 +19,79 @@
 // Write a function called firstItem that passes the first item of the given array to the callback function
 
 const foods = ['pineapple', 'mango', 'ribeye', 'curry', 'tacos', 'ribeye', 'mango'];
+const firstItem = (arr, cb) => {
+  return cb(arr[0]);
+};
 
 firstItem(foods, (firstItem) => {
   console.log(`The first item is ${firstItem}.`);
 });
 
 // Write a function called getLength that passes the length of the array into the callback
+const getLength = (arr, cb) => {
+  return cb(arr.length);
+};
 
 getLength(foods, (length) => {
   console.log(`The length of the array is ${length}.`);
 });
 
 // Write a function called last which passes the last item of the array into the callback
-
+const last = (arr, cb) => {
+  cb(arr[arr.length - 1]);
+};
 last(foods, (lastItem) => {
   console.log(`The last item in the array is ${lastItem}.`);
 });
 
 // Write a function called sumNums that adds two numbers and passes the result to the callback
-
-
+const sumNums = (x, y, cb) => {
+  cb(x + y);
+};
 sumNums(5, 10, (sum) => {
   console.log(`The sum is ${sum}.`);
 });
 
 // Write a function called multiplyNums that adds two numbers and passes the result to the callback
-
+const multiplyNums = (x, y, cb) => {
+  return cb(x * y);
+};
 multiplyNums(5, 10, (product) => {
   console.log(`The product is ${product}.`);
 });
 
 // Write a function called contains that checks if an item is present inside of the given array.
 // Pass true to the callback if it is, otherwise pass false
-
+const contains = (arr, item, cb) => {
+  return cb(arr.includes(item));
+};
 contains(foods, 'ribeye', (result) => {
   console.log(result ? 'ribeye is in the array' : 'ribeye is not in the array');
 });
 
 // Write a function called removeDuplicates that removes all duplicate values from the given array.
 // Pass the array to the callback function.  Do not mutate the original array.
-
+// const removeDuplicates = (arr, cb) => {
+//   const set = new Set(arr);
+//   console.log(set);
+//   return cb(set);
+// };
+const removeDuplicates = (arr, cb) => {
+  const newArray = arr.filter((item, index, inputArray) => {
```

Comment:
Nice! This is a very common solution for removing duplicates. I love filter. It has been very useful to me throughout my career.. as well as indexOf.